L1Trigger/L1THGCalUtilities/plugins/ntuples/HGCalTriggerNtupleHGCMulticlusters.cc

```diff
@@ -3,60 +3,71 @@
 #include "L1Trigger/L1THGCal/interface/HGCalTriggerGeometryBase.h"
 #include "L1Trigger/L1THGCalUtilities/interface/HGCalTriggerNtupleBase.h"
 #include "L1Trigger/L1THGCal/interface/backend/HGCalTriggerClusterIdentificationBase.h"
+#include "DataFormats/L1THGCal/interface/HGCalClusterT.h"
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerTools.h"
 
-class HGCalTriggerNtupleHGCMulticlusters : public HGCalTriggerNtupleBase {
-public:
-  HGCalTriggerNtupleHGCMulticlusters(const edm::ParameterSet& conf);
-  ~HGCalTriggerNtupleHGCMulticlusters() override{};
-  void initialize(TTree&, const edm::ParameterSet&, edm::ConsumesCollector&&) final;
-  void fill(const edm::Event& e, const edm::EventSetup& es) final;
-
-private:
-  void clear() final;
-
-  edm::EDGetToken multiclusters_token_;
-
-  std::unique_ptr<HGCalTriggerClusterIdentificationBase> id_;
-
-  int cl3d_n_;
-  std::vector<uint32_t> cl3d_id_;
-  std::vector<float> cl3d_pt_;
-  std::vector<float> cl3d_energy_;
-  std::vector<float> cl3d_eta_;
-  std::vector<float> cl3d_phi_;
-  std::vector<int> cl3d_clusters_n_;
-  std::vector<std::vector<uint32_t>> cl3d_clusters_id_;
-  // cluster shower shapes
-  std::vector<int> cl3d_showerlength_;
-  std::vector<int> cl3d_coreshowerlength_;
-  std::vector<int> cl3d_firstlayer_;
-  std::vector<int> cl3d_maxlayer_;
-  std::vector<float> cl3d_seetot_;
-  std::vector<float> cl3d_seemax_;
-  std::vector<float> cl3d_spptot_;
-  std::vector<float> cl3d_sppmax_;
-  std::vector<float> cl3d_szz_;
-  std::vector<float> cl3d_srrtot_;
-  std::vector<float> cl3d_srrmax_;
-  std::vector<float> cl3d_srrmean_;
-  std::vector<float> cl3d_emaxe_;
-  std::vector<float> cl3d_bdteg_;
-  std::vector<int> cl3d_quality_;
+class HGCalTriggerNtupleHGCMulticlusters : public HGCalTriggerNtupleBase
+{
+
+  public:
+    HGCalTriggerNtupleHGCMulticlusters(const edm::ParameterSet& conf);
+    ~HGCalTriggerNtupleHGCMulticlusters() override{};
+    void initialize(TTree&, const edm::ParameterSet&, edm::ConsumesCollector&&) final;
+    void fill(const edm::Event& e, const edm::EventSetup& es) final;
+
+  private:
+    void clear() final;
+
+    edm::EDGetToken multiclusters_token_;
+
+    std::unique_ptr<HGCalTriggerClusterIdentificationBase> id_;
+
+    HGCalTriggerTools triggerTools_;
+
+    int cl3d_n_ ;
+    std::vector<uint32_t> cl3d_id_;
+    std::vector<float> cl3d_pt_;
+    std::vector<float> cl3d_energy_;
+    std::vector<float> cl3d_eta_;
+    std::vector<float> cl3d_phi_;
+    std::vector<int> cl3d_clusters_n_;
+    std::vector<std::vector<uint32_t>> cl3d_clusters_id_;
+    std::vector<std::vector<float_t>> cl3d_clusters_layer_pt_;
```

Comment:
Use `float` instead of `float_t` for consistency with other variables.
`cl3d_layers_pt_` would be a better name since the granularity is not the clusters inside the multi-cluster